### DISCUSSION TITLE
gap boundary sequence influenced GapHeatTransfer result

### DISCUSSION THREAD

**Discussion by BoZeng1997** [idaholab/moose#21759] [Q&A Modules: General] [answered]
Hi,

I am doing a 2d steady thermal diffusion simulation with gap radiation on a two separated quarter annuluses. I found that the sequence I specify the `primary` and `secondary` boundary has an influence on the resulting temperature field. `gap_surface_0` as `primary` the simulation did not converge. Specifying two `GapHeatTransfer` for `0-1` and `1-0`  respectively resulted in low temperature at the corners as shown in the figure. However, if I only specify `1` as `primary` and `0` as `secondary` (which matches the direction of heat flux across the gap), no corner temperature issue. 
From my shallow understanding of the code, this object is written to consider heat transfer from both sides. So something was not right about my implementation. Please help me on it. 
![2pellet_01_10_noTdt_illus](https://user-images.githubusercontent.com/59585661/182473597-98dba1c3-df38-47a4-b9c7-0850a55cb159.png)

**Comment by BoZeng1997** [comment, marked as answer]
```
[GlobalParams]
  temperature = T
[]

[Mesh]
  [ori]
    type = FileMeshGenerator
    file = 'radiation_2d.msh'
  []
  [drl]
    type = BlockDeletionGenerator
    input = ori
    block = 'void_center void_pellet'
  []
[]

[Variables]
  [T]
    [InitialCondition]
      type = ConstantIC
      value = 23.0
    []
  []
[]

[Kernels]
  [diff]
    type = ADHeatConduction #ADHeatConduction
    variable = 'T'
    thermal_conductivity = thermal_conductivity
    block = 'pellet_inner pellet_outer'
  []
[]

[BCs]
  [outer_temperature]
    type = ADDirichletBC
    variable = T
    boundary = void_nb_0
    value = 1304.2 # (C)
  []
  [inner_temperature]
    type = ADDirichletBC
    variable = T
    boundary = void_center
    value = 1001.2 # (C)
  []
[]

[ThermalContact]
  # [gap_pellet_01]
  #   type = GapHeatTransfer
  #   variable = T
  #   primary = void_pellet_0
  #   secondary = void_pellet_1
  #   emissivity_primary = 0.37
  #   emissivity_secondary = 0.37
  #   gap_conductivity = 0.151e-3
  #   gap_geometry_type = CYLINDER
  #   cylinder_axis_point_1 = '0 0 0'
  #   cylinder_axis_point_2 = '0 0 1'
  # []
  [gap_pellet_10]
    type = GapHeatTransfer
    variable = T
    primary = void_pellet_1
    secondary = void_pellet_0
    emissivity_primary = 0.37
    emissivity_secondary = 0.37
    gap_conductivity = 0.151e-3
    gap_geometry_type = CYLINDER
    cylinder_axis_point_1 = '0 0 0'
    cylinder_axis_point_2 = '0 0 1'
  []
[]


[Materials]
  [pellet_thermal_conductivity]
    type = ADParsedMaterial
    args = 'T'
    f_name = 'thermal_conductivity'
    function = '250.6e-2' #HELP CONVERGE
    block = 'pellet_inner pellet_outer'
  []
[]

[Postprocessors]
  [void_1_temp]
    type = SideAverageValue
    variable = T
    boundary = void_pellet_1
  []
  [void_0_temp]
    type = SideAverageValue
    variable = T
    boundary = void_pellet_0
  []
  [void_1_flux]
    type = ADSideDiffusiveFluxIntegral
    variable = T
    boundary = void_pellet_1
    diffusivity = thermal_conductivity
  []
  [void_0_flux]
    type = ADSideDiffusiveFluxIntegral
    variable = T
    boundary = void_pellet_0
    diffusivity = thermal_conductivity
  []
[]

[Executioner]
  type = Steady
  solve_type = PJFNK
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
  automatic_scaling = true

  nl_rel_tol = 1e-10
  nl_abs_tol = 1e-8
[]

[Outputs]
  [exodus]
    type = Exodus
    file_base = 2pellet_gap_10_noTdt_cylinder
  []
  print_linear_residuals = false
[]
```

**Reply by lindsayad** [reply]
@jasondhales do you have any suggestions for this question?

**Reply by jasondhales** [reply]
Without `quadrature = true`, this will be a one way (secondary to primary, if I recall correctly) contact constraint.  With `quadrature = true`, the primary and secondary surfaces are used twice, swapped the second time.  This is what we recommend.

That being said, the mortar approach is better.

**Reply by BoZeng1997** [reply]
By mortar approach, do you mean `ModularGapConductanceConstraint` together with `GapFluxModelRadiation` as mentioned in page
https://mooseframework.inl.gov/source/constraints/ModularGapConductanceConstraint.html
and example input deck
https://github.com/idaholab/moose/blob/b3df9cac31d90926bc845ad48628b9b8e0bbb115/modules/heat_conduction/test/tests/gap_heat_transfer_mortar/large_gap_heat_transfer_test_cylinder_mortar.i
?

**Reply by lindsayad** [reply]
yes `ModularGapConductanceConstraint` used with various options for gap flux models (including the radiation one you mentioned) are MOOSE's thermal mortar contact objects

**Comment by BoZeng1997** [comment]
Hi @jasondhales ,

I have another question about simulation gap radiation heat transfer in MOOSE. Does `GapHeatTransfer` or the mortar approach work well for gaps with changing distance? Like the gaps on both sides of the green region in the image. Is this shape fine with the code or the theory of gap radiation?
![fluted_gaps](https://user-images.githubusercontent.com/59585661/186356891-d79c38db-c348-42e0-97ac-56f3a953f870.png)

**Reply by lindsayad** [reply]
thermal contact is generally used in conjunction with mechanics modeling and mesh deformation, so the answer is that MOOSE's thermal models should work fine with changing gaps
